src/overlays/actors/ovl_En_Ganon_Mant/z_en_ganon_mant.c

```diff
@@ -21,19 +27,444 @@ const ActorInit En_Ganon_Mant_InitVars = {
+        Matrix_RotateY(BINANG_TO_RAD(this->actor.shape.rot.y), MTXMODE_NEW);
+        Matrix_MultVec3f(&delta, &posStep);
+        for (i = 0; i < GANON_MANT_NUM_JOINTS; i++) {
+            (pos + i)->x += posStep.x;
+            (pos + i)->z += posStep.z;
+        }
+        // Set length
+        jointLength = 6.5f;
+    } else {
+        jointLength = 9.5f;
+    }
+
+    for (i = 0; i < GANON_MANT_NUM_JOINTS; i++, pos++, vel++, rot++, nextPos++) {
+        if (i == 0) {
+            // Constraint: first position is always root
+            pos->x = root->x;
+            pos->y = root->y;
+            pos->z = root->z;
+        } else {
+            // Decelerate
+            Math_ApproachZeroF(&vel->x, 1.0f, 0.1f);
+            Math_ApproachZeroF(&vel->y, 1.0f, 0.1f);
+            Math_ApproachZeroF(&vel->z, 1.0f, 0.1f);
+
+            // Push the cloak away from attached actor, plus oscillations
+            delta.x = 0;
+            delta.z = (this->backPush + (sinf((strandNum * (2 * M_PI)) / 2.1f) * this->backSwayMagnitude)) *
+                      sBackSwayCoefficients[i];
+            Matrix_RotateY(this->baseYaw, MTXMODE_NEW);
+            Matrix_MultVec3f(&delta, &backSwayOffset);
+
+            // Push the cloak out to either side, in a swaying manner
+            delta.x = cosf((strandNum * M_PI) / (GANON_MANT_NUM_STRANDS - 1.0f)) * this->sideSwayMagnitude *
+                      sSideSwayCoefficients[i];
+            delta.z = 0;
+            Matrix_MultVec3f(&delta, &sideSwayOffset);
+
+            // Calculate position difference
+            gravity = this->gravity;
+            x = ((pos->x + vel->x) - (pos - 1)->x) + (backSwayOffset.x + sideSwayOffset.x);
+            y = ((pos->y + vel->y) - (pos - 1)->y) + gravity;
+            z = ((pos->z + vel->z) - (pos - 1)->z) + (backSwayOffset.z + sideSwayOffset.z);
+
+            // Calculate rotations in the direction of the position difference
+            yaw = Math_Atan2F(z, x);
+            x = -Math_Atan2F(sqrtf(SQ(x) + SQ(z)), y);
+            (rot - 1)->x = x;
+
+            // Calculate real position difference of correct length in the correct direction
+            delta.x = 0;
+            delta.z = jointLength;
+            Matrix_RotateY(yaw, MTXMODE_NEW);
+            Matrix_RotateX(x, MTXMODE_APPLY);
+            Matrix_MultVec3f(&delta, &posStep);
+
+            // Save position
+            x = pos->x;
+            y = pos->y;
+            z = pos->z;
+
+            // Calculate next position
+            pos->x = (pos - 1)->x + posStep.x;
+            pos->y = (pos - 1)->y + posStep.y;
+            pos->z = (pos - 1)->z + posStep.z;
+
+            // Pushes the cloak away from the actor if it is too close
+            if (sqrtf(SQ(pos->x - this->actor.world.pos.x) + SQ(pos->z - this->actor.world.pos.z)) <
+                (sDistMultipliers[i] * this->minDist)) {
+                yaw = Math_Atan2F(pos->z - this->actor.world.pos.z, pos->x - this->actor.world.pos.x);
```

Comment:
This change and making the pad a s32[4] should fix the non matching:
```suggestion
            xDiff = pos->x - this->actor.world.pos.x;
            zDiff = pos->z - this->actor.world.pos.z;
            if (sqrtf(SQ(xDiff) + SQ(zDiff)) < (sDistMultipliers[i] * this->minDist)) {
                yaw = Math_Atan2F(zDiff, xDiff);
```